### ISSUE TITLE
Linter doesn't raise any issues

### ISSUE THREAD

**Issue by cscanlin** [AtomLinter/linter-flake8#172]
Hi, I continually see "No Issues" listed at the bottom of my editor. I'm sure my flake8 is successfully installed and that the executable path for it in settings, matches what I see when I enter `which flake8` (~/anaconda/bin/flake8).

Is it possible that this issue is with the base linter package? What else could be causing this error?

Thanks

**Comment by Arcanemagus** [comment]
Do you have any other linter plugins installed? Do any of them work?

If so, can you open Atom's Developer Tools and set a breakpoint [here](https://github.com/AtomLinter/linter-flake8/blob/v1.13.0/lib/main.coffee#L182) to check whether the `lint()` function is even being called?

**Comment by cscanlin** [comment]
I will try another linter now and see if that helps, but it appears that `lint()` is being called. I've attached a screenshot with some extra debug output from stepping through that might be helpful. Let me know if I can help provide any other details.

<img width="1476" alt="screen shot 2016-03-23 at 9 56 32 am" src="https://cloud.githubusercontent.com/assets/2954573/13993475/cf289f3a-f0dd-11e5-8dcf-487ea79252b3.png">

**Comment by cscanlin** [comment]
I was able to get pylint to work, so this problem is confirmed to be specific to flake8.

**Comment by mattdarveniza** [comment]
I'm also getting this. JSHint and ESLint working fine in Atom, flake8 working fine on the command line (with no venvs or anything like that), getting no lint reporting for flake8

**Comment by Arcanemagus** [comment]
It looks like the parameters being sent to it are correct, if you place a breakpoint [here](https://github.com/AtomLinter/linter-flake8/blob/v1.13.0/lib/main.coffee#L207) what does the `result` object contain?

**Comment by andialbrecht** [comment]
I noticed a similar behavior: when the flake8 executable couldn't be found I see an error message in a popup window. But if it is found the linter doesn't show any errors.

The `result` object looks like this at the above mentioned breakpoint:

```
Object {stdout: "", stderr: ""}
```

**Comment by andialbrecht** [comment]
FTR I noticed that flake8 outputs nothing when called directly on the command line with the same file. pyflakes shows errors as expected. So maybe this isn't an issue with linter-flake8 but flake8 itself.

**Comment by andialbrecht** [comment]
I got tricked by https://github.com/SublimeLinter/SublimeLinter-flake8/issues/20

A `# flake8: noqa` comment at the end of a line disables flake8 for the whole file (and not only a single line). After replacing this specific comment with `# noqa` flake8 and linter-flake8 worked as expected. Sorry for the noise ;)

**Comment by leorochael** [comment]
I also had a Python file with "No Issues" even though it clearly had flake8 issues. Setting the breakpoint recomended by @Arcanemagus above didn't stop the execution, which told me that the linter was not being called at all.

Looking at the preferences for the base Linter package I found the setting "Do Not Lint Files Ignored by VCS" switched on, and the file I wanted to scan was in a section of my repository that was ignored by my top-level project git repository (because it was managed by a nested git repository which is fetched by the build tool)

As a quick reminder, if you have a complex repository layout (e.g. using zc.buildout) where some of the files you edit are ignored by the top level git directory, remember to switch off that setting.

**Comment by Arcanemagus** [comment]
Nice catch @leorochael!

@cscanlin Are you still having issues?

**Comment by SvanT** [comment]
I got the same problem and for me the issue was that i had format=pylint in my ~/.config/flake8 file so the linter-flake8-regex did not work. This can be fixed in linter-flake8 by always adding --format=default to the spawn parameters so it always get the expected format.

**Comment by Arcanemagus** [comment]
Heh, #217 didn't actually "fix" this, but I'm going to leave this as closed since @cscanlin hasn't responded.
